avocado/utils/crypto.py

```diff
@@ -33,9 +39,10 @@ def hash_file(filename, size=None, algorithm="md5"):
     :param filename: Path of the file that will have its hash calculated.
     :param algorithm: Method used to calculate the hash (default is md5).
     :param size: If provided, hash only the first size bytes of the file.
+    :param buffer_size: Buffer size for reading file data.
+    :param num_threads: Number of threads to use for parallel processing.
     :return: Hash of the file, if something goes wrong, return None.
     """
-    chunksize = io.DEFAULT_BUFFER_SIZE
```

Comment:
Can you please split the work into two commits, one for buffer size change and the second for concurrent computing, and write down the description of those changes into commit messages. You can use what you wrote in the PR description. With such changes will be much easier to search in commit history and understand what the changes do.